Fix moveFile for Android

karolba opened the pull request:
Since Android 6.0 calling creating a hardlink is [forbidden by design](https://stackoverflow.com/questions/32365690/is-android-m-not-allowing-hard-links). Every call to `link()` returns `EACCESS`. Therefore, projects such as [Termux](https://termux.com) are having problem with using npm to install packages on Android. This is reported in termux/termux-packages#1192 and npm/npm#17480. After manually patching cacache in npm with this change I was able to succesfully install packages from npm on Android.

This pull requests fixes the issue, by making cacache move files using `move-concurrently` if `link()` fails also on Android, just as it was  intended to work on `win32`.

As I was testing it, I've noticed the path where `link()` failed on `win32` was also not functioning as intended, throwing `ENOENT` every time. That is also addressed in this PR.

Commit:
Fix a typo, moveFile would always fail without link() available

Commit:
Add android to systems which can have a non-working link()

espeed commented:
This fix works, and I was able to install firebase-tools on Chromebook via:

`npm install -g firebase-tools `

zkat committed:
remove special case here

this was a premature optimization. Let `move-concurrently` do its damn job on its own terms, here.

zkat reviewed (approved):
UGH. After looking into it, I think this is why we've been having issues on a bunch of platforms. The fallback literally never worked.

This was a wonderful catch! Thank you so much.

zkat commented:
This patch is available in the npm canary, and will go out with the next release of npm, btw. You can use the canary with: `npx npmc install`, or just install it globally with `npm i -g npmc`
